source/common/upstream/cluster_manager_impl.cc

```diff
@@ -179,13 +179,15 @@ void ClusterManagerImpl::initializeClustersFromV1Json(const Json::Object& config
 }
 
 void ClusterManagerImpl::initializeClustersFromV2Proto(const envoy::api::v2::Bootstrap& bootstrap) {
-  for (const auto& cluster : bootstrap.bootstrap_clusters()) {
-    loadCluster(cluster, false);
+  if (bootstrap.has_static_resources()) {
+    for (const auto& cluster : bootstrap.static_resources().clusters()) {
+      loadCluster(cluster, false);
+    }
   }
 
   // We can now potentially create the CDS API once the backing cluster exists.
-  if (bootstrap.has_cds_config()) {
-    cds_api_ = factory_.createCds(bootstrap.cds_config(), sds_config_, *this);
+  if (bootstrap.has_dynamic_resources() && bootstrap.dynamic_resources().has_cds_config()) {
```

Comment:
nit: same here with first clause. Will stop commenting now for other things. @htuch is out right now but looking at other code I think this is what he has been doing. You can also see: https://github.com/lyft/envoy/pull/1521/files for some of the code he started to write. (That change needs mega merging so might as well just continue with this one since it's much smaller).